hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/DotNode.java

```diff
@@ -583,39 +577,44 @@ public FromElement getImpliedJoin() {
 	}
 
 	/**
-	 * Is the given property name a reference to the primary key of the associated
-	 * entity construed by the given entity type?
+	 * Is the given property name a reference to the join key of the associated
+	 * entity constructed by the given entity type?
+	 * <p/>
+	 *
+	 * This method resolves the {@code propertyName} as an property of the entity type at the
+	 * {@link #propertyPath} relative to the {@link #getFromElement() FromElement}.
+	 * The implementation does so by invoking {@link EntityPersister#getPropertyType(String)},
+	 * which will resolve the property path against the entitiy's {@link org.hibernate.persister.entity.PropertyMapping}.
+	 * On initialisation of the {@link EntityPersister}, this {@code PropertyMapping} is filled with
```

Comment:
BTW, minor nitpick and no need to change it here but as it seems you want to contribute a lot: we usually use American English so it would be `initialization` and `optimized`. Not sure everything in the code is consistent with it tbh but that's a good rule of thumb.